### PR TITLE
Try to fix multiple season snatches. #4201 #4168

### LINKED ISSUE #4168: double downloading episodes and whole seasons
### Before submitting your issue:

Enable debug logging in Medusa settings, reproduce the error (be sure to disable after the bug is fixed)

**Branch/Commit: master
**OS: windows 10
**What you did: this is in reference to a continuance of this previous issue: https://github.com/pymedusa/Medusa/issues/4105 and i added several new shows.
**What happened: basically episodes are getting downloaded twice or more.
**What you expected: shows to only be downloaded once.
**Logs:**
```
[application2.log](https://github.com/pymedusa/Medusa/files/1978449/application2.log)
[application.log](https://github.com/pymedusa/Medusa/files/1978452/application.log)
```

### LINKED ISSUE #4201: Multiple Snatches of Season pack from different providers
Just had a really weird thing happen where Medusa snatched a season pack for The Terror on 5 different providers.

Medusa Info: | Branch: master                 Commit: 77b20916ab577b82112ebc624f816054788c63f2                 Version: v0.2.2                 Database: 44.9
-- | --
Python Version: | 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)]
SSL Version: | OpenSSL 1.0.2k  26 Jan 2017
OS: | Windows-10-10.0.14393
Locale: | nl_NL.cp1252

```
2018-05-15 17:48:10 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Downloading The.Terror.S01.1080p.WEB.H264-DEFLATE with 20 seeders and 2 leechers and size 1.89 GB from Nebulance
2018-05-15 17:48:09 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Downloading The.Terror.S01.1080p.WEB.H264-DEFLATE.[NO.RAR] with 266 seeders and 21 leechers and size 29.80 GB from IPTorrents
2018-05-15 17:48:08 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Downloading The.Terror.S01.1080p.AMZN.WEB-DL.DD+.5.1.H.264-DEFLATE with 84 seeders and 9 leechers and size 29.90 GB from PrivateHD
2018-05-15 17:48:07 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Downloading The.Terror.S01.1080p.AMZN.WEB-DL.DDP5.1.H.264-DEFLATE with 121 seeders and 7 leechers and size 0.00 B from HDTorrents
2018-05-15 17:48:06 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Downloading The.Terror.S01.1080p.WEB.H264-DEFLATE.[NO.RAR] with 129 seeders and 15 leechers and size 29.89 GB from TorrentDay
2018-05-15 17:48:06 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Nebulance] :: [77b2091] Quality of The.Terror.S01.1080p.WEB.H264-DEFLATE is 1080p WEB-DL
2018-05-15 17:48:06 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Nebulance] :: [77b2091] Adding multi-ep result for full-season torrent. Undesired episodes can be skipped in torrent client if desired!
2018-05-15 17:48:06 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Nebulance] :: [77b2091] Quality of The.Terror.S01.720p.AMZN.WEBRip.DDP5.1.x264-KiTTeN is 720p WEB-DL
2018-05-15 17:48:06 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Nebulance] :: [77b2091] Quality of The.Terror.S01.1080p.WEB.H264-DEFLATE is 1080p WEB-DL
2018-05-15 17:48:05 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Nebulance] :: [77b2091] Performing season pack search for The Terror
2018-05-15 17:48:05 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Torrentz2] :: [77b2091] Quality of The.Terror.S01E09.www.subtorrent.net.avi is Unknown
2018-05-15 17:48:05 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Torrentz2] :: [77b2091] Quality of The.Terror.S01E08.www.subtorrent.net.avi is Unknown
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Torrentz2] :: [77b2091] Performing episode search for The Terror
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Quality of The.Terror.S01E09.WEBRip.x264-ION10 is SDTV
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Quality of The.Terror.S01E09.WEB.H264-DEFLATE[rartv] is SDTV
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Quality of The.Terror.S01E09.720p.WEB.H264-DEFLATE[rartv] is 720p WEB-DL
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Quality of The.Terror.S01E09.1080p.WEB.H264-DEFLATE[rartv] is 1080p WEB-DL
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Quality of The.Terror.S01E08.Terror.Camp.Clear.HDTV.x264-CRiMSON[rartv] is SDTV
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Quality of The.Terror.S01E08.Terror.Camp.Clear.720p.HDTV.x264-CRiMSON[rartv] is 720p HDTV
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Quality of The.Terror.S01E08.WEB.H264-DEFLATE[rartv] is SDTV
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Quality of The.Terror.S01E08.720p.WEB.H264-DEFLATE[rartv] is 720p WEB-DL
2018-05-15 17:47:45 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Quality of The.Terror.S01E08.1080p.WEB.H264-DEFLATE[rartv] is 1080p WEB-DL
2018-05-15 17:47:34 INFO     SEARCHQUEUE-BACKLOG-322191 :: [Rarbg] :: [77b2091] Performing episode search for The Terror
2018-05-15 17:47:34 INFO     SEARCHQUEUE-BACKLOG-322191 :: [IPTorrents] :: [77b2091] Quality of The.Terror.S01.1080p.WEB.H264-DEFLATE.[NO.RAR] is 1080p WEB-DL
2018-05-15 17:47:34 INFO     SEARCHQUEUE-BACKLOG-322191 :: [IPTorrents] :: [77b2091] Adding multi-ep result for full-season torrent. Undesired episodes can be skipped in torrent client if desired!
2018-05-15 17:47:33 INFO     SEARCHQUEUE-BACKLOG-322191 :: [IPTorrents] :: [77b2091] Quality of The.Terror.S01.1080p.WEB.H264-DEFLATE.[NO.RAR] is 1080p WEB-DL
2018-05-15 17:47:33 INFO     SEARCHQUEUE-BACKLOG-322191 :: [IPTorrents] :: [77b2091] Ignoring The.Terror.S01.WEBRip.1080p.DD5.1.H265-d3g based on ignored words filter: h265
2018-05-15 17:47:33 INFO     SEARCHQUEUE-BACKLOG-322191 :: [IPTorrents] :: [77b2091] Quality of The.Terror.S01.WEBRip.1080p.DD5.1.H265-d3g is 1080p WEB-DL
2018-05-15 17:47:29 INFO     SEARCHQUEUE-BACKLOG-322191 :: [IPTorrents] :: [77b2091] Performing season pack search for The Terror
2018-05-15 17:47:12 INFO     SEARCHQUEUE-BACKLOG-322191 :: [TorrentLeech] :: [77b2091] Performing episode search for The Terror
2018-05-15 17:47:03 INFO     SEARCHQUEUE-BACKLOG-322191 :: [TorrentLeech] :: [77b2091] Performing season pack search for The Terror
2018-05-15 17:47:03 INFO     SEARCHQUEUE-BACKLOG-322191 :: [PrivateHD] :: [77b2091] Preferring The.Terror.S01.1080p.AMZN.WEB-DL.DD+.5.1.H.264-DEFLATE (preferred words)
2018-05-15 17:47:03 INFO     SEARCHQUEUE-BACKLOG-322191 :: [PrivateHD] :: [77b2091] Quality of The.Terror.S01.1080p.AMZN.WEB-DL.DD+.5.1.H.264-DEFLATE is 1080p WEB-DL
2018-05-15 17:47:03 INFO     SEARCHQUEUE-BACKLOG-322191 :: [PrivateHD] :: [77b2091] Adding multi-ep result for full-season torrent. Undesired episodes can be skipped in torrent client if desired!
2018-05-15 17:47:03 INFO     SEARCHQUEUE-BACKLOG-322191 :: [PrivateHD] :: [77b2091] Quality of The.Terror.S01.720p.AMZN.WEBRip.DD+.5.1.x264-KiTTeN is 720p WEB-DL
2018-05-15 17:47:03 INFO     SEARCHQUEUE-BACKLOG-322191 :: [PrivateHD] :: [77b2091] Preferring The.Terror.S01.1080p.AMZN.WEB-DL.DD+.5.1.H.264-DEFLATE (preferred words)
2018-05-15 17:47:03 INFO     SEARCHQUEUE-BACKLOG-322191 :: [PrivateHD] :: [77b2091] Quality of The.Terror.S01.1080p.AMZN.WEB-DL.DD+.5.1.H.264-DEFLATE is 1080p WEB-DL
2018-05-15 17:47:02 INFO     SEARCHQUEUE-BACKLOG-322191 :: [PrivateHD] :: [77b2091] Performing season pack search for The Terror
2018-05-15 17:47:02 INFO     SEARCHQUEUE-BACKLOG-322191 :: [HDTorrents] :: [77b2091] Preferring The.Terror.S01.1080p.AMZN.WEB-DL.DDP5.1.H.264-DEFLATE (preferred words)
2018-05-15 17:47:02 INFO     SEARCHQUEUE-BACKLOG-322191 :: [HDTorrents] :: [77b2091] Quality of The.Terror.S01.1080p.AMZN.WEB-DL.DDP5.1.H.264-DEFLATE is 1080p WEB-DL
2018-05-15 17:47:02 INFO     SEARCHQUEUE-BACKLOG-322191 :: [HDTorrents] :: [77b2091] Adding multi-ep result for full-season torrent. Undesired episodes can be skipped in torrent client if desired!
2018-05-15 17:47:02 INFO     SEARCHQUEUE-BACKLOG-322191 :: [HDTorrents] :: [77b2091] Quality of The.Terror.S01.720p.AMZN.WEBRip.DDP5.1.x264-KiTTeN is 720p WEB-DL
2018-05-15 17:47:02 INFO     SEARCHQUEUE-BACKLOG-322191 :: [HDTorrents] :: [77b2091] Preferring The.Terror.S01.1080p.AMZN.WEB-DL.DDP5.1.H.264-DEFLATE (preferred words)
2018-05-15 17:47:02 INFO     SEARCHQUEUE-BACKLOG-322191 :: [HDTorrents] :: [77b2091] Quality of The.Terror.S01.1080p.AMZN.WEB-DL.DDP5.1.H.264-DEFLATE is 1080p WEB-DL
2018-05-15 17:47:00 INFO     SEARCHQUEUE-BACKLOG-322191 :: [HDTorrents] :: [77b2091] Performing season pack search for The Terror
2018-05-15 17:47:00 INFO     SEARCHQUEUE-BACKLOG-322191 :: [TorrentDay] :: [77b2091] Quality of The.Terror.S01.1080p.WEB.H264-DEFLATE.[NO.RAR] is 1080p WEB-DL
2018-05-15 17:47:00 INFO     SEARCHQUEUE-BACKLOG-322191 :: [TorrentDay] :: [77b2091] Adding multi-ep result for full-season torrent. Undesired episodes can be skipped in torrent client if desired!
2018-05-15 17:47:00 INFO     SEARCHQUEUE-BACKLOG-322191 :: [TorrentDay] :: [77b2091] Quality of The.Terror.S01.1080p.WEB.H264-DEFLATE.[NO.RAR] is 1080p WEB-DL
2018-05-15 17:47:00 INFO     SEARCHQUEUE-BACKLOG-322191 :: [TorrentDay] :: [77b2091] Quality of The.Terror.S01.WEBRip.x264-NO is SDTV
2018-05-15 17:47:00 INFO     SEARCHQUEUE-BACKLOG-322191 :: [TorrentDay] :: [77b2091] Ignoring The.Terror.S01.WEBRip.1080p.DD5.1.H265-d3g based on ignored words filter: h265
2018-05-15 17:47:00 INFO     SEARCHQUEUE-BACKLOG-322191 :: [TorrentDay] :: [77b2091] Quality of The.Terror.S01.WEBRip.1080p.DD5.1.H265-d3g is 1080p WEB-DL
2018-05-15 17:46:58 INFO     SEARCHQUEUE-BACKLOG-322191 :: [TorrentDay] :: [77b2091] Performing season pack search for The Terror
2018-05-15 17:46:58 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Using backlog search providers
2018-05-15 17:46:58 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Building internal name cache for The Terror
2018-05-15 17:46:58 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Finished processing 2780 scene exceptions.
2018-05-15 17:46:58 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Updating exception_cache and exception_season_cache
2018-05-15 17:46:58 INFO     SEARCHQUEUE-BACKLOG-322191 :: [77b2091] Beginning backlog search for: The Terror
```

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

---

- Should fix #4201 
- Should fix #4168

### CODE REVIEW

File: medusa/search/core.py [hunk 1]
```diff
@@ -819,8 +819,8 @@ def search_providers(series_obj, episodes, forced_search=False, down_cur_quality
 
                 if not multi_needed_eps:
                     log.debug(
-                        u'All of these episodes were covered by another multi-episode nzb, '
-                        u'ignoring this multi-ep result'
+                        u'All of these episodes were covered by another multi-episode,'
```
Comment (sharkykh) [inline medusa/search/core.py]: Maybe this sounds better?
> All of these episodes were covered by another multi-episode **result**, ignoring this multi-ep result

File: medusa/search/core.py [hunk 2]
```diff
@@ -830,13 +830,12 @@ def search_providers(series_obj, episodes, forced_search=False, down_cur_quality
                     if ep_obj.episode in found_results[cur_provider.name]:
                         log.debug(
                             u'A needed multi-episode result overlaps with a single-episode result for episode {0},'
-                            u' removing the single-episode results from the list',
+                            u' replacing the single-episode results from the list',
                             ep_obj.episode,
                         )
-                        del found_results[cur_provider.name][ep_obj.episode]
+                    found_results[cur_provider.name][ep_obj.episode] = multi_result
```
Comment (sharkykh) [inline medusa/search/core.py]: I'm not sure, but could this cause the same `multi_result` to appear multiple times (once for each episode)?

Comment (medariox) [reply medusa/search/core.py]: In theory that's what this code should prevent: https://github.com/pymedusa/Medusa/blob/3a769ad89776bf9b40f3905539725bc8237f5140/medusa/search/core.py#L788-L825

Not sure if it actually works tho.

File: medusa/search/core.py (hunk 1 above)
Comment (medariox) [reply medusa/search/core.py]: Sounds better indeed, changed 👍

File: medusa/search/core.py (hunk 2 above)
Comment (sharkykh) [reply medusa/search/core.py]: I read through the code, and I actually think it's [this code](https://github.com/pymedusa/Medusa/blob/d39c1ba/medusa/search/core.py#L855-L867).
Anyway I think that part is fine.
Now I'm not so sure if the change will actually fix anything. :thinking:

**Comment by medariox** [comment]
What do you mean by "I can't make it work"?

**Comment by medariox** [comment]
Included in https://github.com/pymedusa/Medusa/pull/4311

**Comment by Diablobait** [comment]
well on spanish for example, i configure provider to search for season, the website upload for example Prison break T1 and medusa never download it

**Pull request closed without merging**